### ISSUE TITLE
Add SVG material design icons support

### ISSUE THREAD

**Issue by gsans** [tombatossals/angular-openlayers-directive#63]
Nice work! Comming from reddit. I recently did a small experiment with AngularJS and OpenLayers 3. See here: https://medium.com/angularjs-meetup-south-london/angular-integration-with-openlayers-3-5a6e8d29e635. Includes github/slides/demo.

Few featues that you may enjoy:
- SVG Material Design icons - crispy clean markers improving UX on mobiles
- Animations - better UX
- Custom zoom to extent control
- Filtering markers

Maybe the most beneficial is the SVG icons as you can style them (shape, color) and historically it was a pain as you had to create the markers as images.

**Comment by tombatossals** [comment]
That's a really nice piece of code! I'll take a look at it, there are very interesting ideas I could add to this project, if you don't mind.

**Comment by gsans** [comment]
Thanks! Feel free to browse and use what you like. I'm very happy to share
it. There's also a post to explain what I'm doing if it's not clear from
the code. =)

**Comment by nbiton** [comment]
Using SVG assets for achieving crispy looking assets on mobile, is not a great idea.
I tried doing this, and if you have more than a few markers on your map frame at a given point, there's a very significant performance hit and panning movement renders at 3-5 fps.
This is caused by mobile browsers inability to render SVG to canvas efficiently.

I had use raster based markers instead, and the frame rate for the same amount of markers was over 30 fps.

**Comment by gsans** [comment]
Good point. Nonetheless dropping a feature for everybody because there's a
corner scenario that is not very performant (mobile and many markers) is a
little extreme.

SVG icons are generally available on mobile and a great way to avoid having
to manually create and style different image based markers, eg: for each
possible color and size. Besides SVG also allow for extra CSS styling and
animations.

In general for first page load performance, maybe adding Angular plus  your
library to the mix is not the way to go considering you can achieve all
features with plain JavaScript and have Google Closure, with two way data
binding already there.

**Comment by nbiton** [comment]
SVG assets used for markers are rendered to the canvas, therefore, they will be static and couldn't be manipulated with CSS or animations.

I think it might be possible to achieve the same crispy look of SVG on the markers, if the SVG asset is rendered once to a canvas, then .toDataUrl is used to extract the SVG as image and that is set as data URI to the src attribute of the ol.Style.Icon instance, but I haven't attempted that yet.

Initial page load didn't appear to be affected, btw.
The performance hit was noticeable while panning and zooming the map.

**Comment by gsans** [comment]
That's not the implementation I'm using but its good to know about other
approaches.

I was talking about time for first render as you have to download Angular +
library + OpenLayers. That's a big chunk for mobile.

**Comment by nbiton** [comment]
Yeah, I see, but I suggested that it might be a way to achieve the a result which looks the same while working around the performance issue I had.

The file size didn't matter that much for my use case, as I'm using this component in a hybrid app.

**Comment by gsans** [comment]
I see. I thought this was a general discussion. Yeah that is something I've
used before with good results. It also improves zoom in/out in my
experience.

**Comment by MartinDesigner** [comment]
![imghome](https://cloud.githubusercontent.com/assets/13809960/9289329/0bef602c-436c-11e5-80fc-3ec07f82dd8d.jpg)
So, as a supplement, however as a main point try this amazing free app. Createer.com is an online software for easily designing online ads, responsive landing pages and print materials such as flyers, posters, covers, invitations and more. Createer lets you easily share your designs on social media, send them via email, download as HTML, image or PDF. You can even design UI mockups for mobile as for standard display (html, js, css always generating without any coding required from you). Createer also includes 3,000 predesigned templates (browse here: https://beta2.createer.com/Templates?ghi) and over 250,000 images (illustrations, photos and icons) to work with in its user-friendly drag’n’drop enviroment. And finally this amazing stuff is here: https://beta2.createer.com/?ghi
